nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/bulk/TestSetPropertiesAction.java

```diff
@@ -118,8 +118,12 @@ protected void testSetProperties(String username) throws Exception {
         for (DocumentModel child : session.query(nxql)) {
             assertEquals(title, child.getTitle());
             assertEquals(description, child.getPropertyValue("dc:description"));
-            assertEquals(foo, child.getPropertyValue("cpx:complex/foo"));
-            assertEquals(bar, child.getPropertyValue("cpx:complex/bar"));
+            try {
+                assertEquals(foo, child.getPropertyValue("cpx:complex/foo"));
+                assertEquals(bar, child.getPropertyValue("cpx:complex/bar"));
```

Comment:
If I understand correctly, `try/catch` is used to not fail the test when child is a File or a Note.
You should check the document type to do these assertions.